### ISSUE TITLE
common control parameters details need documentation

### ISSUE THREAD

**Issue by jensens** [plone/documentation#590]
great blog about it: http://blog.redturtle.it/2012/07/03/boxing-plone
- `ajax_load=1` is documented in current docs: http://docs.plone.org/develop/addons/javascript/ajax.html?highlight=ajax_load#id4
- `ajax_include_head=1` - missing: we have some great hidden useful feature see 
- `disable_border=1` -> empties toolbar in P5 (except personal-tools)
- `disable_PORTLETMANAGERNAME=1` (see blog post above)

**Comment by stevepiercy** [comment]
@jensens @pnicolli @ale-rt's blog post is gone. There's only a teaser on webarchive.

https://web.archive.org/web/20171204193439/http://blog.redturtle.it/2012/07/03/

I'm going to close this issue as won't fix, but if the content is relevant to Plone 6 and recoverable, then please reopen and paste the content here.

**Comment by ale-rt** [comment]
FTR, the original [shameless self-promotion activated] MASTERPIECE [shameless self-promotion deactivated] can be admired in all its glory here:

- https://web.archive.org/web/20211203011938/http://blog.redturtle.it/boxing-plone

The parameters are still valid for Classic Plone (because classics never go out of style):

- https://classic.demo.plone.org/login_form?ajax_load=1&ajax_include_head=1
- https://classic.demo.plone.org/en/demo?disable_plone.leftcolumn=1

If I'm not mistaken, there’s a parameter to turn off the toolbar too. I am not sure and too lazy to confirm that :)

**Comment by ale-rt** [comment]
I lied, I am not lazy enough:
```
[ale@avo omelette]$ rg disable.*toolbar Products/CMFPlone/browser/ploneview.py
51:        if "disable_border" in request or "disable_toolbar" in request:
```

See https://github.com/plone/Products.CMFPlone/blob/26c659aaad12ce59376d3d79ca12b2d1128fa38a/Products/CMFPlone/browser/ploneview.py#L48-L54

**Comment by stevepiercy** [comment]
Reopening. Thank you, @ale-rt!

Before it disappears forever, I copy-pasted the article here, and updated some links. I didn't proof read it thoroughly, and it will need a some work to make it into documentation.

---

We start from the typical use case of making a Plone page fit inside an iframe, than we'll see the many ways Plone page rendering process can be controlled by properly setting request parameters

## Did you ever wanted...

... to just include the main content from a Plone page in an iframe and for this reason struggled with customized views and CSS rules to remove unwanted page elements like portlets, viewlets, footer and so on?

## Did you know...

... that there is a simple and easy way to request Plone to serve a page with only the main content, skipping all the surrounding elements for navigation?

## For the impatients

To embed a page inside an iframe, since Plone 4.0.3,  you have the option to include just the main content by appending the query string: `?ajax_load=1&ajax_include_head=1`, to the URL inside the `src` attribute.

As an example, with this test HTML:

```html
<html>
 <head></head>
 <body>
   <h1>Boxing Plone</h1>
   <div>
     <iframe width="480" height="200"
             src="http://localhost:8080/Plone/login?ajax_load=1&amp;ajax_include_head=1"></iframe>
   </div>
   <div>
     <iframe width="480" height="260"
             src="http://localhost:8080/Plone/sitemap?ajax_load=1&amp;ajax_include_head=1"></iframe>
   </div>
 </body>
</html>
``` 

You end up with something like this:

[image missing]

The page displays the login form and the sitemap without all the elements that are not relevant for this use case (logo, headers, footer, portlets, viewlets and so on).

## Warnings

Be aware that customizing the style in the "iframed" HTML is possible using some javascript techniques (see for example [this question on stackoverflow](https://stackoverflow.com/questions/217776/how-to-apply-css-to-iframe)) and that if you include HTML from a different domain you may have to struggle with the [Same Origin Policy](https://en.wikipedia.org/wiki/Same_origin_policy).

If you use a customized main template that is not aware of this technique you could have different results!

## tl;dr AKA Toooo Long: Don't Read AKA Hic Sunt Leones

The parts of the page that "disappear", when request parameters are properly set, are not rendered at all by the server because the default main_template is "request aware". Looking at it (see [sunburst main_template.pt on github](https://web.archive.org/web/20211203011938/http://github.com/plone/plonetheme.sunburst/blob/master/plonetheme/sunburst/skins/sunburst_templates/main_template.pt)) it is possible to understand in detail what is happening.

The ajax_load parameter, if set to a non null value, disables the rendering of many sections of the template (e.g. portal_top, viewlets and the portlet columns).

The ajax_include_head forces the plone.head provider to be rendered even if the ajax_load parameter would remove it from the page.

If ajax_load is set and ajax_include_head is not, you will end up with and unstyled HTML page that will also miss the javascript. This behaviour turns out to be handy when making javascript overlays: in this case the styles and javascripts are taken from the page that will start the ajax call.

Since Plone 4, there is also the chance to disable portlet columns by adding the request parameter `?disable_$NAME=1`, where `$NAME` is the portlet manager (e.g., `plone.leftcolumn` or `plone.rightcolumn`). So the URL to disable both the left and right column will be something like this:

```text
http://127.0.0.1:8080/Plone?disable_plone.leftcolumn=1
```

On github you can find the [relevant code](https://web.archive.org/web/20211203011938/http://github.com/plone/plone.app.layout/blob/master/plone/app/layout/globals/layout.py): take a look to the have_portlets method.

In many ZPT templates those request parameters are enforced using dummy defines:


```xml
disable_column_one python:request.set('disable_plone.leftcolumn',1);
disable_column_two python:request.set('disable_plone.rightcolumn',1);
```

Take a look (e.g.) to [the dashboard page template](https://github.com/plone/plone.app.layout/blob/master/plone/app/layout/dashboard/dashboard.pt).

The capability of Plone to disable template parts by request is not a prerogative of Plone 4. For example, for years now, setting the parameter `?disable_border=1`, it is possible to remove the "green bar" on top of the views (the one that appears when you are logged in and let's you add/modify contents and make some more actions).
Do you know...

... any other trick like those?
